com.unity.visualeffectgraph/Editor/GraphView/Views/Properties/Vector2PropertyRM.cs

```diff
@@ -0,0 +1,126 @@
+using UnityEditor.UIElements;
+using UnityEditor.VFX.UI;
+using UnityEngine;
+using UnityEngine.UIElements;
+
+namespace UnityEditor.VFX.UI
+{
+    class Vector2PropertyRM : SimpleUIPropertyRM<Vector2, Vector2>
+    {
+        VFXVector2Field m_VectorField;
+        protected VFXMinMaxSliderField m_Slider;
+
+        public Vector2PropertyRM(IPropertyRMProvider controller, float labelWidth) : base(controller, labelWidth)
+        {}
+
+        private void ValueDragStarted()
+        {
+            m_Provider.StartLiveModification();
+        }
+
+        private void ValueDragFinished()
+        {
+            m_Provider.EndLiveModification();
+            hasChangeDelayed = false;
+            NotifyValueChanged();
+        }
+
+        public override float GetPreferredControlWidth()
+        {
+            return 120;
+        }
+
+        public override INotifyValueChanged<Vector2> CreateField()
+        {
+            INotifyValueChanged<Vector2> result;
+            bool isMinMax = m_Provider.attributes.Is(VFXPropertyAttributes.Type.MinMax);
+            if (isMinMax)
+            {
+                Vector2 range = m_Provider.attributes.FindRange();
+                result = CreateSliderField(out m_Slider);
+                m_Slider.onValueDragFinished = ValueDragFinished;
```

Comment:
May be for consistency you could either move the event registration in the CreateSliderField or move them out of the CreateSimpleField?